Test failures on Windows

Checkstyle fails on every file since we don't specify which line ending is expected. Additionally when running the unit test suite on windows the follow failures occur in `java-manta-client-unshaded` tests. ~IIRC `verifyOrderingWithMoreSubdirectoriesDataSet` generates unicode characters without bounding the allowed codepoints and could be fixed by using `RandomStringGenerator.Builder#withinRange(int, int)`~ The first failure is caused by the way we generate path names (they're not actually ??? but non-printing characters) and needs a modulo applied to the path segment generation . The second failure ~needs additional investigation~ can be fixed by using `File#toURI`.

```
[ERROR] Tests run: 380, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 84.467 s <<< FAILURE! - in TestSuite
[ERROR] verifyOrderingWithMoreSubdirectoriesDataSet(com.joyent.manta.client.MantaObjectDepthComparatorTest)  Time elapsed: 2.437 s  <<< FAILURE!
java.nio.file.InvalidPathException: Illegal char <|> at index 119: /user/stor/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/???/||||.json
        at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
        at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
        at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
        at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
        at java.nio.file.Paths.get(Paths.java:84)
        at com.joyent.manta.client.MantaObjectDepthComparatorTest.assertOrdering(MantaObjectDepthComparatorTest.java:94)
        at com.joyent.manta.client.MantaObjectDepthComparatorTest.verifyOrderingWithMoreSubdirectoriesDataSet(MantaObjectDepthComparatorTest.java:77)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
        at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
        at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
        at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
        at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
        at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
        at org.testng.TestRunner.privateRun(TestRunner.java:744)
        at org.testng.TestRunner.run(TestRunner.java:602)
        at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
        at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
        at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
        at org.testng.SuiteRunner.run(SuiteRunner.java:289)
        at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
        at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
        at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
        at org.testng.TestNG.runSuites(TestNG.java:1144)
        at org.testng.TestNG.run(TestNG.java:1115)
        at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
        at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
        at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:373)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:334)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:119)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:407)

[ERROR] canLoadKeyFromURIPath(com.joyent.manta.client.crypto.SecretKeyUtilsTest)  Time elapsed: 0.031 s  <<< FAILURE!
java.lang.IllegalArgumentException: Illegal character in authority at index 7: file://C:\cygwin64\tmp\ciphertext-1445704935919878175.data
        at java.net.URI.create(URI.java:852)
        at com.joyent.manta.client.crypto.SecretKeyUtilsTest.canLoadKeyFromURIPath(SecretKeyUtilsTest.java:71)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
        at org.testng.internal.Invoker.invokeMethod(Invoker.java:661)
        at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:869)
        at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1193)
        at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:126)
        at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
        at org.testng.TestRunner.privateRun(TestRunner.java:744)
        at org.testng.TestRunner.run(TestRunner.java:602)
        at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
        at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
        at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
        at org.testng.SuiteRunner.run(SuiteRunner.java:289)
        at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
        at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
        at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
        at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
        at org.testng.TestNG.runSuites(TestNG.java:1144)
        at org.testng.TestNG.run(TestNG.java:1115)
        at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
        at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
        at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:373)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:334)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:119)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:407)
Caused by: java.net.URISyntaxException: Illegal character in authority at index 7: file://C:\cygwin64\tmp\ciphertext-1445704935919878175.data
        at java.net.URI$Parser.fail(URI.java:2848)
        at java.net.URI$Parser.parseAuthority(URI.java:3186)
        at java.net.URI$Parser.parseHierarchical(URI.java:3097)
        at java.net.URI$Parser.parse(URI.java:3053)
        at java.net.URI.<init>(URI.java:588)
        at java.net.URI.create(URI.java:850)
        ... 30 more
```

--- a/checkstyle.xml
+++ b/checkstyle.xml
@@ -33,7 +33,9 @@
 
     <!-- Checks whether files end with a new line.                        -->
     <!-- See http://checkstyle.sf.net/config_misc.html#NewlineAtEndOfFile -->
-    <module name="NewlineAtEndOfFile"/>
+    <module name="NewlineAtEndOfFile">
+        <property name="lineSeparator" value="lf"/>
+    </module>
 
     <!-- Checks that property files contain the same keys.         -->
     <!-- See http://checkstyle.sf.net/config_misc.html#Translation -->

--- a/java-manta-client-unshaded/src/test/java/com/joyent/manta/client/MantaObjectDepthComparatorTest.java
+++ b/java-manta-client-unshaded/src/test/java/com/joyent/manta/client/MantaObjectDepthComparatorTest.java
@@ -109,8 +109,9 @@ public class MantaObjectDepthComparatorTest {
 
         for (int i = 0; i < number; i++) {
             final MantaObject object = mock(MantaObject.class);
+            final int segmentChar = (number + i) % 26;
             String path = dirObject.getPath() + SEPARATOR
-                    + StringUtils.repeat((char)(97 + number + i), 4) + ".json";
+                    + StringUtils.repeat(((char)(97 + segmentChar) ), 4) + ".json";
             when(object.getPath()).thenReturn(path);
             when(object.getType()).thenReturn(MantaObject.MANTA_OBJECT_TYPE_OBJECT);
             when(object.getContentType()).thenReturn(ContentType.APPLICATION_JSON.toString());
@@ -129,7 +130,8 @@ public class MantaObjectDepthComparatorTest {
         path.append("stor");
 
         for (int i = 3; i <= depth; i++) {
-            path.append(SEPARATOR).append(StringUtils.repeat((char)(97 + depth + i), 3));
+            final int segmentChar = (depth + i) % 26;
+            path.append(SEPARATOR).append(StringUtils.repeat(((char)(97 + segmentChar) ), 3));
             MantaObject object = mockDirectory(path);
             objects.add(object);
         }

--- a/java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/SecretKeyUtilsTest.java
+++ b/java-manta-client-unshaded/src/test/java/com/joyent/manta/client/crypto/SecretKeyUtilsTest.java
@@ -11,7 +11,6 @@ import org.apache.commons.io.FileUtils;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
-import javax.crypto.SecretKey;
 import java.io.File;
 import java.io.IOException;
 import java.net.URI;
@@ -19,6 +18,7 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Arrays;
+import javax.crypto.SecretKey;
 
 @Test
 public class SecretKeyUtilsTest {
@@ -68,7 +68,7 @@ public class SecretKeyUtilsTest {
         File file = File.createTempFile("ciphertext-", ".data");
         FileUtils.forceDeleteOnExit(file);
         FileUtils.writeByteArrayToFile(file, keyBytes);
-        URI uri = URI.create("file://" + file.getAbsolutePath());
+        URI uri = file.toURI();
 
         SecretKey expected = SecretKeyUtils.loadKey(keyBytes, AesGcmCipherDetails.INSTANCE_128_BIT);
         SecretKey actual = SecretKeyUtils.loadKeyFromPath(Paths.get(uri), AesGcmCipherDetails.INSTANCE_128_BIT);

